### DISCUSSION TITLE
Ctrl-2 enters "2" in text, doesn't send Ctrl-2 to application

### DISCUSSION THREAD

**Discussion by emendelson** [The-Wineskin-Project/WineskinServer#272] [General]
This is a very minor question, but I wonder if there is a fix.

I have a Wineskin app that runs the vDos MS-DOS emulator, which runs the DOS word-processor WordPerfect. WordPerfect uses all four states of the function keys, F1, Ctrl-F1, Alt-F1, Shift-F1. In the macOS System Preferences, Keyboard Shortcuts, I've disabled everything in macOS that uses those keys, and they all work perfectly in my app.

The one problem is this: WordPerfect uses Ctrl-2 to create accented characters. You type Ctrl-2, then (for example) "e" and the single-quote key, and this enters e-acute. This works perfectly when I run vDos under Windows, and it works perfectly when I run WordPerfect in the macOS version of DOSBox. But when I type Ctrl-2 in my Wineskin app, it produces only a "2" in the text.

WordPerfect also uses Ctrl-6 to switch keyboard layouts. Again, this works perfectly under Windows and in DOSBox, but not in my Wineskin app.

Ctrl-letter keys work correctly. It's only Ctrl-number keys that don't work.

This is NOT affected by the two mapping options in the Advanced setting in Wineskin: it makes no difference if Map Option is Alt and Map Command as Ctrl are checked - Ctrl-number doesn't work.

Is this something that's likely to be an upstream issue, or is there any way to check on keyboard mappings in Wineskin? I can live with this problem if there's no alternative, but it would be nice to be able to fix it.

**Comment by Gcenx** [comment]
From what you’ve explained this sounds like an upstream problem.

**Comment by emendelson** [comment]
OK, I'll try to work out a bug report for winehq.

**Reply by Gcenx** [reply]
The usually thing with upstream is they’ll want you to run using the latest wine-devel so do you have a 64Bit application your familiar with?

Pre-build upstream packages can be install via brews cask system those simply install packages I’m providing. So that would at least save you a headache.

**Comment by emendelson** [comment]
Yes, I've got a 64-bit version of vDos that I can use for testing. And thank you for the headache relief!

**Comment by emendelson** [comment]
Posted: https://bugs.winehq.org/show_bug.cgi?id=53243

**Reply by Gcenx** [reply]
I’d assume the component would be `winemac.drv`

**Comment by emendelson** [comment]
I should have realized that. Many years ago, I figured out how to patch that component. Will try to get a look at the source code in case there's anything that might be fixable, but it's almost certainly far beyond my abilities.

**Comment by emendelson** [comment]
I remembered that the file is dlls/winemac.drv/keyboard.c, and the relevant part of the code seems to be around lines 1600-1700, more or less. But I don't see anything that would make Ctrl-letters work correct but Ctrl-numbers not work correctly. Probably I'm missing something obvious.
